idl.md

```diff
@@ -1,61 +1,90 @@
-## Proposed IDL
-```javascript
+# Proposed IDL
+
+```js
 partial interface ServiceWorkerRegistration {
-  readonly attribute SyncManager syncManager;
+  readonly attribute SyncManager sync;
+  readonly attribute PeriodicSyncManager periodicSync;
 };
 
 interface SyncManager {
   Promise<SyncRegistration> register(optional SyncRegistrationOptions options);
-  Promise<SyncRegistration> getRegistration(DOMString id);
+  Promise<SyncRegistration> getRegistration(DOMString tag);
   Promise<sequence<SyncRegistration>> getRegistrations();
-  Promise<SyncPermissionStatus> hasPermission();
-  readonly attribute unsigned long minAllowablePeriod;
+  Promise<SyncPermissionState> permissionState();
```

Comment:
The reasoning was to align with push, which I think is still a good reason to keep the longer names.